test/Microsoft.Azure.Devices.Edge.Test.Common/Platform.cs

```diff
@@ -13,11 +18,57 @@ public static Task<string> CollectDaemonLogsAsync(DateTime testStartTime, string
             ? Windows.EdgeDaemon.CollectLogsAsync(testStartTime, filePrefix, token)
             : Linux.EdgeDaemon.CollectLogsAsync(testStartTime, filePrefix, token);
 
-        // TODO: download installer script from aka.ms if user doesn't pass installerPath in Windows
         public static IEdgeDaemon CreateEdgeDaemon(Option<string> installerPath) => IsWindows()
-            ? new Windows.EdgeDaemon(installerPath.Expect(() => new ArgumentException()))
+            ? new Windows.EdgeDaemon(installerPath)
             : new Linux.EdgeDaemon() as IEdgeDaemon;
 
+        // After calling this function, the following files will be available under {scriptPath}:
+        //  certs/iot-edge-device-{deviceId}-full-chain.cert.pem
+        //  private/iot-edge-device-{deviceId}.key.pem
+        public static Task<EdgeCertificates> GenerateEdgeCertificatesAsync(string deviceId, string scriptPath, CancellationToken token) => IsWindows()
```

Comment:
Since we are adding more methods in this static class, should it be better to use factory pattern here; introduce a Platform interface; and it will be easier to extend to more platforms in the future.  What do you think?

Reply:
It certainly is getting heavier. I'll take a closer look at refactoring it.

Reply:
I introduced an `IPlatform` interface, and implemented it for Linux and Windows. Have a look...